### ISSUE TITLE
Error while building: temporary value dropped while borrowed

### ISSUE THREAD

**Issue by Tiebe** [mb1986/qrex#1]
Hi! When trying to build the project, the following two errors appear:

```rust
error[E0716]: temporary value dropped while borrowed
  --> src/resource/runner.rs:52:14
   |
51 |         let name = if offset > 0 {
   |             ---- borrow later stored here
52 |             &self.read_name(reader, record.name_offset)?
   |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ creates a temporary value which is freed while still in use
53 |         } else {
   |         - temporary value is freed at the end of this statement
   |
   = note: consider using a `let` binding to create a longer lived value

error[E0716]: temporary value dropped while borrowed
  --> src/main.rs:65:14
   |
62 |           let base_path = if args.skip_dirs {
   |               --------- borrow later stored here
...
65 |               &config
   |  ______________^
66 | |                 .output_path
67 | |                 // FIXME: use below implementation instead of current one
68 | |                 //.join(format!("{:#x}_{:#x}_{:#x}", tree, names, data))
...  |
71 | |                     res.addresses.tree, res.addresses.names, res.addresses.data
72 | |                 ))
   | |__________________^ creates a temporary value which is freed while still in use
73 |           };
   |           - temporary value is freed at the end of this statement
   |
   = note: consider using a `let` binding to create a longer lived value

For more information about this error, try `rustc --explain E0716`.
error: could not compile `qrex` (bin "qrex") due to 2 previous errors
```

**Comment by mb1986** [comment]
Hi @Tiebe, could you please share the Rust compiler version you currently use? (`rustc --version` or `rustup --version`)

It compiles nicely on the latest stable Rust 1.82.0:
```text
$ rustc --version
rustc 1.82.0 (f6e511eec 2024-10-15)
mb@hive: ~ 
$ rustup --version
rustup 1.27.1 (2024-05-07)
info: This is the version for the rustup toolchain manager, not the rustc compiler.
info: The currently active `rustc` version is `rustc 1.82.0 (f6e511eec 2024-10-15)`
```

You could also try `rustup update` first to update the Rust compiler.

Thank you for raising this, I'll add information of the lowest needed Rust version.

**Comment by mb1986** [comment]
I've just checked the minimum supported rust version (using `crate msrv find`) and the result is below:

```text
   Considered (min … max):   Rust 1.56.1 … Rust 1.82.0 
   Search method:            bisect                    
   MSRV:                     1.79.0                    
   Target:                   x86_64-unknown-linux-gnu  
```

So to compile `qrex` you have to have at least Rust **1.79.0**.

**Comment by mb1986** [comment]
I've just released **0.3.0** with even lower rust version requirement:

```text
   Considered (min … max):   Rust 1.56.1 … Rust 1.82.0 
   Search method:            bisect                    
   MSRV:                     1.74.1                    
   Target:                   x86_64-unknown-linux-gnu  
```

Please try this one without any other changes to the rust version in case you are on **1.74.1** or above.

**Comment by mb1986** [comment]
For the reference, this is a change which solves your error in 1.79 and higher: https://blog.rust-lang.org/2024/06/13/Rust-1.79.0.html#extending-automatic-temporary-lifetime-extension.
